### PR TITLE
CHE-1419: add ability to run simple java project which has several so…

### PR DESCRIPTION
**Pull request by svor**
…urce folders

Improved the command for running Java project with several source folders.
@vparfonov take a look pls!

### CODE REVIEW

File: plugins/plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/command/valueproviders/SourcepathProvider.java [hunk 1]
```diff
@@ -64,39 +64,35 @@ public String getKey() {
...
+        final String projectPath = project.get().getLocation().toString();
 
-                                classpath.append(':');
+        return classpathContainer.getClasspathEntries(projectPath).then(new Function<List<ClasspathEntryDto>, String>() {
+            @Override
+            public String apply(List<ClasspathEntryDto> arg) throws FunctionException {
+                classpathResolver.resolveClasspathEntries(arg);
+                Set<String> sources = classpathResolver.getSources();
+                StringBuilder classpath = new StringBuilder("");
+                for (String source : sources) {
```
Comment (vparfonov) [inline plugins/plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/command/valueproviders/SourcepathProvider.java:86]: may be just new StringBuilder() ?

File: plugins/plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/command/valueproviders/ClasspathProvider.java [hunk 2]
```diff
@@ -65,38 +65,36 @@ public String getKey() {
...
 
-        return promises.resolve("");
+        final String projectPath = project.get().getLocation().toString();
+
+        return classpathContainer.getClasspathEntries(projectPath).then(new Function<List<ClasspathEntryDto>, String>() {
+            @Override
+            public String apply(List<ClasspathEntryDto> arg) throws FunctionException {
+                classpathResolver.resolveClasspathEntries(arg);
+                Set<String> libs = classpathResolver.getLibs();
+                StringBuilder classpath = new StringBuilder("");
```
Comment (vparfonov) [inline plugins/plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/command/valueproviders/ClasspathProvider.java]: way you use StringBuilder with "" ?

File: plugins/plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/command/valueproviders/SourcepathProvider.java (hunk 1 above)
Comment (svor) [inline plugins/plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/command/valueproviders/SourcepathProvider.java:86]: you are right!

**Comment by vparfonov** [comment]
ok

**Pull request merged**